File: RunSeason2/backend/the_largiest_square/square.py
def largest_square(n, m, grid):
    dp = [[0] * m for _ in range(n)]
    max_side = 0
    max_i = 0
    max_j = 0

    for i in range(n):
        for j in range(m):
            if grid[i][j] == 1:
                if i == 0 or j == 0:
                    dp[i][j] = 1
                else:
                    dp[i][j] = min(dp[i-1][j], dp[i][j-1], dp[i-1][j-1]) + 1
                if dp[i][j] > max_side:
                    max_side = dp[i][j]
                    max_i = i
                    max_j = j
                elif dp[i][j] == max_side:
                    if j > max_j or (j == max_j and i > max_i):
                        max_i = i
                        max_j = j

    top_left_i = max_i - max_side + 1
    top_left_j = max_j - max_side + 1

    print(max_side)
    print(top_left_i + 1, top_left_j + 1)

def main():
    n, m = map(int, input().split())
    grid = []
    for _ in range(n):
        line = list(map(int, input().split()))
        grid.append(line)
    largest_square(n, m, grid)

if __name__ == '__main__':
    main()

File: RunSeason2/backend/the_largiest_square/square.go
package main

import (
	"fmt"
)

func largestSquare(n, m int, grid [][]int) {
	dp := make([][]int, n)
	for i := range dp {
		dp[i] = make([]int, m)
	}

	maxSide := 0
	maxI := 0
	maxJ := 0

	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			if grid[i][j] == 1 {
				if i == 0 || j == 0 {
					dp[i][j] = 1
				} else {
					dp[i][j] = min(dp[i-1][j], dp[i][j-1], dp[i-1][j-1]) + 1
				}

				if dp[i][j] > maxSide {
					maxSide = dp[i][j]
					maxI = i
					maxJ = j
				} else if dp[i][j] == maxSide {
					if j > maxJ || (j == maxJ && i > maxI) {
						maxI = i
						maxJ = j
					}
				}
			}
		}
	}

	// координаты верхнего левого угла квадрата
	topLeftI := maxI - maxSide + 1
	topLeftJ := maxJ - maxSide + 1

	// добавляем единицу к координатам, чтобы привести их к формату ввода-вывода (1-индексация)
	fmt.Println(maxSide)
	fmt.Println(topLeftI+1, topLeftJ+1)
}

func min(a, b, c int) int {
	if a < b {
		if a < c {
			return a
		}
		return c
	}
	if b < c {
		return b
	}
	return c
}

func main() {
	var n, m int
	fmt.Scan(&n, &m)

	grid := make([][]int, n)
	for i := range grid {
		grid[i] = make([]int, m)
		for j := 0; j < m; j++ {
			fmt.Scan(&grid[i][j])
		}
	}

	largestSquare(n, m, grid)
}
